File: izysafe/flutter/lib/core/token_store.dart
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

/// Persists the parent JWT pair (access 24h / refresh 30d) in secure storage.
class TokenStore {
  static const _access = 'izysafe.access_token';
  static const _refresh = 'izysafe.refresh_token';

  final FlutterSecureStorage _storage;
  TokenStore([FlutterSecureStorage? storage])
      : _storage = storage ?? const FlutterSecureStorage();

  // In-memory mirror so the request interceptor stays synchronous and fast.
  String? _accessCache;
  String? _refreshCache;

  Future<void> load() async {
    _accessCache = await _storage.read(key: _access);
    _refreshCache = await _storage.read(key: _refresh);
  }

  String? get access => _accessCache;
  String? get refresh => _refreshCache;
  bool get hasSession => _accessCache != null;

  Future<void> save(String access, String refresh) async {
    _accessCache = access;
    _refreshCache = refresh;
    await _storage.write(key: _access, value: access);
    await _storage.write(key: _refresh, value: refresh);
  }

  Future<void> clear() async {
    _accessCache = null;
    _refreshCache = null;
    await _storage.delete(key: _access);
    await _storage.delete(key: _refresh);
  }
}

File: izysafe/web-admin/src/lib/tokenStore.ts
/**
 * Persists the school-admin JWT pair in localStorage.
 *
 * Access token (24h) rides on every request; the refresh token (30d) is used
 * only by the refresh interceptor. Rotation replaces both on each refresh.
 */
import type { TokenPair } from "@/types/api";

const ACCESS_KEY = "izysafe.access_token";
const REFRESH_KEY = "izysafe.refresh_token";

export const tokenStore = {
  getAccess(): string | null {
    return localStorage.getItem(ACCESS_KEY);
  },
  getRefresh(): string | null {
    return localStorage.getItem(REFRESH_KEY);
  },
  set(tokens: Pick<TokenPair, "access_token" | "refresh_token">): void {
    localStorage.setItem(ACCESS_KEY, tokens.access_token);
    localStorage.setItem(REFRESH_KEY, tokens.refresh_token);
  },
  clear(): void {
    localStorage.removeItem(ACCESS_KEY);
    localStorage.removeItem(REFRESH_KEY);
  },
  hasSession(): boolean {
    return Boolean(localStorage.getItem(ACCESS_KEY));
  },
};

/** Broadcast a forced logout so the AuthProvider can redirect to /login. */
export const AUTH_LOGOUT_EVENT = "izysafe:auth:logout";

export function emitForcedLogout(): void {
  window.dispatchEvent(new Event(AUTH_LOGOUT_EVENT));
}
